docs: fix code style

Piyush132000 opened the pull request:
I have removed unnecessary code from the components & I initialize the form in the OnInit function (life cycle). I think initialization of forms and getting data from the service must be done inside ngOnInit.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines:https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

Update  the code style of Product-list & Shipping component.
Removed Unnecessary code.


<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Same behavior as previous

Issue Number: N/A


## What is the new behavior?
NA
## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

AlirezaEbrahimkhani commented:
@Piyush132000 
Tnx
but can you please update the commit message to more closely follow angular [commit message guidelines](https://github.com/angular/angular/blob/655507012b7b59c83e27429349ce3ac10608a82d/CONTRIBUTING.md#commit)? :pray:
Somthing like this :

> docs: fix code style
> 
> Updated Product list & shipping component code style

 
To update, your last commit message you can use this command `git commit --amend` and then force push your branch
remember you do not need to add another commit message just edit your last commit message. :+1:

Piyush132000 commented:
> @Piyush132000 Tnx but can you please update the commit message to more closely follow angular [commit message guidelines](https://github.com/angular/angular/blob/655507012b7b59c83e27429349ce3ac10608a82d/CONTRIBUTING.md#commit)? 🙏 Somthing like this :
> 
> > docs: fix code style
> > Updated Product list & shipping component code style
> 
> To update, your last commit message you can use this command `git commit --amend` and then force push your branch remember you do not need to add another commit message just edit your last commit message. 👍

@AlirezaEbrahimkhani thanks sir and I updated the commit message.

jessicajaniuk commented:
@Piyush132000 this has some lint and test failures that should be quick to resolve. If you can get them addressed, this is otherwise green and should be quick to land.

Piyush132000 commented:
> @Piyush132000 this has some lint and test failures that should be quick to resolve. If you can get them addressed, this is otherwise green and should be quick to land.

Sorry for inconvenience, can you please help me how can I remove these failure?

Piyush132000 commented:
> @Piyush132000 The test_aio failure looks like a simple linting issue where you need a space after a `:` in one of your files. The test_docs_examples test looks like an issue with one of the types used in the template. You can always click on "Details" in the failed tests listed below in the PR here to see the details of the failed tests.

Thanks @jessicajaniuk

Piyush132000 commented:
Hii @jessicajaniuk, I have fixed lint-related issues. 

Please verify.

jessicajaniuk commented:
@Piyush132000 Looks like the lint issues are fixed, but you still have some test_docs_examples issues. You can see them [here](https://circleci.com/gh/angular/angular/1149963?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

Piyush132000 commented:
> @Piyush132000 Looks like the lint issues are fixed, but you still have some test_docs_examples issues. You can see them [here](https://circleci.com/gh/angular/angular/1149963?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

Sorry for the inconvenience, @jessicajaniuk I tried but was not able to fix the test failed.

Piyush132000 commented:
> > @Piyush132000 Looks like the lint issues are fixed, but you still have some test_docs_examples issues. You can see them [here](https://circleci.com/gh/angular/angular/1149963?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).
> 
> Sorry for the inconvenience, @jessicajaniuk I tried but was not able to fix the failed tests.

Hii @jessicajaniuk  can you please help me more so I can fix these failures.

jessicajaniuk commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -11,15 +11,18 @@ import { CartService } from '../cart.service';
+export class ShippingComponent implements OnInit {
 
-  shippingCosts = this.cartService.getShippingPrices();
+  shippingCosts: {  type: string , price: number}[] = [];
```
This is what's breaking the tests. shippingCosts should be an observable since the cartService `getShippingPrices` method returns an observable. The error in the tests is this:

```
error TS2740: Type 'Observable<{ type: string; price: number; }[]>' is missing the following properties from type '{ type: string; price: number; }[]': length, pop, push, concat, and 27 more
```

This is essentially saying there's a type mismatch between what you've specified here, which is just a simple array of objects, and the actual return value, which is an Observable. This needs to be resolved to fix the tests.

gkalpak commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -11,15 +12,18 @@ import { CartService } from '../cart.service';
+export class ShippingComponent implements OnInit {
 
-  shippingCosts = this.cartService.getShippingPrices();
+  shippingCosts: Observable<{  type: string , price: number}[]>;
```
Nit:

```suggestion
  shippingCosts: Observable<{ type: string, price: number }[]>;
```

gkalpak commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -11,15 +12,18 @@ import { CartService } from '../cart.service';
 // #docregion inject-cart-service
   constructor(private cartService: CartService) { }
 // #enddocregion inject-cart-service
 
+   ngOnInit(): void {
+    this.shippingCosts =  this.cartService.getShippingPrices();
+   }
 
 // #docregion props
```
This should be part of the `props` docregion:

```suggestion
// #docregion props

  ngOnInit(): void {
    this.shippingCosts =  this.cartService.getShippingPrices();
  }

```

Could you also please update the part of the guide where this snippet is used to reflect the change (i.e. mention that the property is initialized in an `ngOnInit()` method)? 🙏 

[1]: https://github.com/angular/angular/blob/42289f25c69ec7a36d2e3ef424d32f77bc7cd19a/aio/content/start/start-data.md?plain=1#L269-L271

gkalpak reviewed (changes requested):
Thx for making these improvements, @Piyush132000 :+1:
I left a couple of suggestions.

Piyush132000 commented:
Hii @jessicajaniuk @gkalpak  please check my PR again. 

Thanks

gkalpak commented on aio/content/start/start-data.md:
```diff
@@ -266,7 +266,7 @@ This section guides you through modifying the `ShippingComponent` to retrieve sh
     <code-example header="src/app/shipping/shipping.component.ts" path="getting-started/src/app/shipping/shipping.component.ts" region="inject-cart-service"></code-example>
 
-1.  Define a `shippingCosts` property that sets the `shippingCosts` property using the `getShippingPrices()` method from the `CartService`.
+1. Define a `shippingCosts` property that sets the `shippingCosts` property using the `getShippingPrices()` method from the `CartService`. Initialize the `shippingCosts` property inside `ngOnInit()` method.
```
Some formatting suggestions (to align with overall style):

```suggestion
1.  Define a `shippingCosts` property that sets the `shippingCosts` property using the `getShippingPrices()` method from the `CartService`.
    Initialize the `shippingCosts` property inside `ngOnInit()` method.
```

mary-poppins commented:
You can preview 052aa44 at https://pr45527-052aa44.ngbuilds.io/.
You can preview 228febc at https://pr45527-228febc.ngbuilds.io/.
You can preview 2068330 at https://pr45527-2068330.ngbuilds.io/.
You can preview c7fb01d at https://pr45527-c7fb01d.ngbuilds.io/.
You can preview 96281dc at https://pr45527-96281dc.ngbuilds.io/.
You can preview cc445b9 at https://pr45527-cc445b9.ngbuilds.io/.

gkalpak commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -11,15 +12,18 @@ import { CartService } from '../cart.service';
 // #docregion inject-cart-service
   constructor(private cartService: CartService) { }
 // #enddocregion inject-cart-service
 
-
 // #docregion props
+   ngOnInit(): void {
```
In order for all docregions to be formatted correctly, we need to have an empty line before `ngOnInit()` (inside the `props` docregion). So, this needs to be:

```diff
 // #enddocregion inject-cart-service
+// #docregion props
 
-// #docregion props
   ngOnInit(): void {
```

gkalpak commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -11,15 +12,18 @@ import { CartService } from '../cart.service';
 // #docregion props
+   ngOnInit(): void {
+    this.shippingCosts =  this.cartService.getShippingPrices();
+   }
```
Incorrect indentation:

```suggestion
  }
```

gkalpak commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -1,8 +1,9 @@
 // #docplaster
 // #docregion imports
-import { Component } from '@angular/core';
+import { Component , OnInit  } from '@angular/core';
```
Nit: Inconsistent spacing:

```suggestion
import { Component, OnInit  } from '@angular/core';
```

gkalpak reviewed (changes requested):
Just a couple of formatting suggestions. Otherwise lgtm :+1:

mary-poppins commented:
You can preview 0da8181 at https://pr45527-0da8181.ngbuilds.io/.

Piyush132000 commented:
Hii @gkalpak @jessicajaniuk, please review my PR.

gkalpak reviewed (approved):
Thx, @Piyush132000 :+1:

gkalpak commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -1,8 +1,9 @@
+import { Component, OnInit } from '@angular/core';
 
 import { CartService } from '../cart.service';
+import { Observable } from 'rxjs';
```
Nit: Put this above the `CartService` import (since we try to keep relative imports below non-relative ones).

gkalpak commented on aio/content/examples/getting-started/src/app/shipping/shipping.component.ts:
```diff
@@ -11,15 +12,18 @@ import { CartService } from '../cart.service';
+export class ShippingComponent implements OnInit {
 
-  shippingCosts = this.cartService.getShippingPrices();
+  shippingCosts: Observable<{ type: string, price: number }[]>;
```
Actually, this change causes TypeScript to complain (and CI to fail) with:

```
Property 'shippingCosts' has no initializer and is not definitely assigned in the constructor
```

Maybe use:

```suggestion
  shippingCosts!: Observable<{ type: string, price: number }[]>;
```

Piyush132000 replied:
Hii @gkalpak , Thanks for the suggestions, I have done your suggested changes. If any other changes are required let me know.

Thanks

jessicajaniuk commented:
@Piyush132000 I rebased this on main for you, but it looks like you still have some failing tests.

Piyush132000 commented:
Hii @jessicajaniuk , @gkalpak  is there any changes required for merge to main?

gkalpak commented:
@Piyush132000, there are some CI failures that need to be fixed. In particular, the `getting-started` example is failing.

You can see the failed test output [here][1]. In case it helps, here is the error message:

```
Warning: 1 rules skipped due to selector errors:
  router-outlet + * -> Cannot read properties of undefined (reading 'type')


Error: src/app/shipping/shipping.component.html:3:52 - error TS2769: No overload matches this call.
  Overload 1 of 3, '(obj: Observable<any[] | Iterable<any> | undefined> | Subscribable<any[] | Iterable<any> | undefined> | Promise<any[] | Iterable<any> | undefined>): any[] | ... 2 more ... | undefined', gave the following error.
    Argument of type '{ type: string; price: number; }[]' is not assignable to parameter of type 'Observable<any[] | Iterable<any> | undefined> | Subscribable<any[] | Iterable<any> | undefined> | Promise<any[] | Iterable<any> | undefined>'.
      Type '{ type: string; price: number; }[]' is missing the following properties from type 'Observable<any[] | Iterable<any> | undefined>': source, operator, lift, subscribe, and 2 more.
  Overload 2 of 3, '(obj: null | undefined): null', gave the following error.
    Argument of type '{ type: string; price: number; }[]' is not assignable to parameter of type 'null | undefined'.
  Overload 3 of 3, '(obj: Observable<any[] | Iterable<any> | undefined> | Subscribable<any[] | Iterable<any> | undefined> | Promise<any[] | Iterable<any> | undefined> | null | undefined): any[] | ... 2 more ... | undefined', gave the following error.
    Argument of type '{ type: string; price: number; }[]' is not assignable to parameter of type 'Observable<any[] | Iterable<any> | undefined> | Subscribable<any[] | Iterable<any> | undefined> | Promise<any[] | Iterable<any> | undefined> | null | undefined'.
      Type '{ type: string; price: number; }[]' is not assignable to type 'Observable<any[] | Iterable<any> | undefined>'.

3 <div class="shipping-item" *ngFor="let shipping of shippingCosts | async">
                                                     ~~~~~~~~~~~~~

  src/app/shipping/shipping.component.ts:10:16
    10   templateUrl: './shipping.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component ShippingComponent.


Error: src/app/shipping/shipping.component.ts:24:5 - error TS2740: Type 'Observable<{ type: string; price: number; }[]>' is missing the following properties from type '{ type: string; price: number; }[]': length, pop, push, concat, and 27 more.

24     this.shippingCosts =  this.cartService.getShippingPrices();
       ~~~~~~~~~~~~~~~~~~
```

[1]: https://circleci.com/gh/angular/angular/1165448

mary-poppins commented:
You can preview 4cb7ec8 at https://pr45527-4cb7ec8.ngbuilds.io/.

Piyush132000 commented:
Hi @jessicajaniuk, @gkalpak Please review my PR.

Thanks

gkalpak commented on aio/content/start/start-data.md:
```diff
@@ -266,7 +266,8 @@ This section guides you through modifying the `ShippingComponent` to retrieve sh
     <code-example header="src/app/shipping/shipping.component.ts" path="getting-started/src/app/shipping/shipping.component.ts" region="inject-cart-service"></code-example>
 
-1.  Define a `shippingCosts` property that sets the `shippingCosts` property using the `getShippingPrices()` method from the `CartService`.
+1. Define a `shippingCosts` property that sets the `shippingCosts` property using the `getShippingPrices()` method from the `CartService`.
+   Initialize the `shippingCosts` property inside `ngOnInit()` method.
```
Nit: The indentation does not match the other items here, but not worth blocking this PR over that.

gkalpak reviewed (approved):
LGTM 🚀
Thx, @Piyush132000 :+1:

Reviewed-for: global-docs-approvers

jessicajaniuk reviewed (approved):
LGTM :cookie:

Thanks for your work and persistence in getting this green, @Piyush132000! 🎉

alxhub reviewed (changes requested):
Hello @Piyush132000 

Sorry to kick this back, but when I was going to merge this I noticed this PR is composed of 4 commits, all of which have the same generic title and no description.

Please squash these commits into one, and add a short description describing why we're making these changes.

Piyush132000 committed:
docs: fix code style

I have removed unnecessary code from the components & i initialize the
form in OnInit function (life cycle). I think initilization of forms and
getting data from service must be done inside ngOnInit.

mary-poppins commented:
You can preview 77979d9 at https://pr45527-77979d9.ngbuilds.io/.

mary-poppins commented:
You can preview 425a619 at https://pr45527-425a619.ngbuilds.io/.

Piyush132000 commented:
Hii @jessicajaniuk, @gkalpak and @alxhub are any changes required to merge this PR to main?. I already squashed the commits.

jessicajaniuk commented:
This PR was merged into the repository by commit 9ce733aacb00777bc1cf01bb9ef1ce9534437512.

Piyush132000 commented:
Thanks, @jessicajaniuk @gkalpak @alxhub
